pkg/security/tests/setsockopt_test.go

```diff
@@ -0,0 +1,101 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux && functionaltests
+
+// Package tests holds tests related files
+package tests
+
+import (
+	"fmt"
+	"syscall"
+	"testing"
+	"unsafe"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DataDog/datadog-agent/pkg/security/secl/model"
+	"github.com/DataDog/datadog-agent/pkg/security/secl/rules"
+)
+
+func TestSetSockOpt(t *testing.T) {
+	SkipIfNotAvailable(t)
+
+	ruleDefs := []*rules.RuleDefinition{
+		{
+			ID:         "test_rule_setsockopt",
+			Expression: `setsockopt.level == SOL_SOCKET && setsockopt.optname == SO_ATTACH_FILTER`,
+		},
+	}
+
+	test, err := newTestModule(t, nil, ruleDefs)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer test.Close()
+
+	t.Run("setsockopt", func(t *testing.T) {
+		var fd int
+		type SockFilter struct {
+			Code uint16
+			Jt   uint8
+			Jf   uint8
+			K    uint32
+		}
+
+		type SockFprog struct {
+			Len    uint16
+			_      [6]byte
+			Filter *SockFilter
+		}
+
+		defer func() {}()
+
+		test.WaitSignal(t, func() error {
+			var err error
+			fd, err = syscall.Socket(syscall.AF_INET, syscall.SOCK_RAW, syscall.IPPROTO_TCP)
```

Comment:
nit: you could do a `fd, err := ` instead

Reply:
I don't think so because he wants to set the value to the fd from the outer scope. Using `:=` would create a new variable in this tight scope and make the test fail